### PR TITLE
Added basic create, edit, delete tests, RBAC tests for HPA

### PR DESCRIPTION
**Pull request by soumyalj**
Please review

### CODE REVIEW

File: tests/validation/tests/v3_api/test_hpa.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+
+    hpa = p_client.create_horizontalPodAutoscaler(
+        name=name,
+        namespaceId=ns.id,
+        maxReplicas=5,
+        minReplicas=2,
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    wait_for_hpa_to_active(p_client, hpa)
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_hpa.py:363]: Should we use the hpa object returned by this method to do the following checks?

File: tests/validation/tests/v3_api/test_hpa.py [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+def test_rbac_hpa_list(remove_resource, role):
+    user_token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    user_client = get_project_client_for_token(project, user_token)
+    hpa = create_hpa(user_client, ns)
+    hpaname = hpa["name"]
+    hpadict = user_client.list_horizontalPodAutoscaler(name=hpaname)
+    print(hpadict)
+    assert hpa.type == "horizontalPodAutoscaler"
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_hpa.py:81]: it is better to check the object returned by the list method, not the one returned by the create method.

File: tests/validation/tests/v3_api/test_hpa.py [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+    with pytest.raises(ApiError) as e:
+        readonly_user_client.update(hpa,
+                                    name=hpa['name'],
+                                    namespaceId=ns.id,
+                                    maxReplicas=10,
+                                    minReplicas=3,
+                                    metrics=metrics)
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'Forbidden'
+
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_hpa.py:276]: remove the hpa at the end of the method

File: tests/validation/tests/v3_api/test_hpa.py [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+    )
+    wait_for_hpa_to_active(p_client, hpa)
+    assert hpa.type == "horizontalPodAutoscaler"
+    assert hpa.name == name
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA object
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
+
+    return hpa
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_hpa.py:371]: Should we return the workload as well? So we can remove both hpa and the workload later.

File: tests/validation/tests/v3_api/test_hpa.py (hunk 1 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_hpa.py:363]: Done

File: tests/validation/tests/v3_api/test_hpa.py (hunk 2 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_hpa.py:81]: Done

File: tests/validation/tests/v3_api/test_hpa.py [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    hpa = wait_for_hpa_to_active(p_client, hpa)
+    assert hpa.type == "horizontalPodAutoscaler"
+    assert hpa.name == name
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA object
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_hpa.py:397]: @soumyalj Can you assert the value of hpa.minReplicas is set to expected value right after it gets created ?

Comment (sangeethah) [reply tests/validation/tests/v3_api/test_hpa.py:397]: Validate that the workload actually has correct number of pods by calling validate_workload

File: tests/validation/tests/v3_api/test_hpa.py [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+        minReplicas=2,
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    wait_for_hpa_to_active(p_client, hpa)
+
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_hpa.py:435]: Validate that the workload actually has correct number of pods by calling validate_workload

File: tests/validation/tests/v3_api/test_hpa.py [hunk 7]
```diff
@@ -0,0 +1,533 @@
...
+    hpa, workload = edit_hpa(p_client, ns)
+    p_client.delete(hpa, workload)
+
+
+def test_delete_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa, workload = create_hpa(p_client, ns)
+    delete_hpa(p_client, hpa, ns)
+    p_client.delete(workload)
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_hpa.py:27]: Can you include cases relating to deleting hpa that are still associated with workloads ?

Comment (soumyalj) [reply tests/validation/tests/v3_api/test_hpa.py:27]: As discussed, its not required

File: tests/validation/tests/v3_api/test_hpa.py (hunk 5 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_hpa.py:397]: Done

File: tests/validation/tests/v3_api/test_hpa.py (hunk 6 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_hpa.py:435]: Done

**Pull request merged**